### PR TITLE
Set file versions in release workflow

### PR DESCRIPTION
**Pull request by lemccomb**
File versions are updated in the release workflow to ensure that the file version matches the release version.

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -270,6 +270,35 @@ jobs:
...
+        # Get all non-test .csproj files in the solution directory and its subdirectories
+        $csprojFiles = Get-ChildItem -Path $env:GITHUB_WORKSPACE -Recurse -Filter "*.csproj"
+          | Where-Object { $_.Name -notmatch "test" }
+
+        foreach ($file in $csprojFiles) {
+          # Read the content of the .csproj file
+          $content = Get-Content -Path $file.FullName
+
+          # Update the version number in the .csproj file
+          $updatedContent = $content -replace '<FileVersion>.*?</FileVersion>', "<FileVersion>$newVersion</FileVersion>"
```
Comment (mayuram2) [inline .github/workflows/dotnet.yml:288]: Should this be $version?

Comment (lemccomb) [reply .github/workflows/dotnet.yml]: When I tried with $version it got interpreted as nuget version. I tested with both and FileVersion is the correct property. But I'm going to try setting it from the publish command so I don't have to modify the project files.

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -273,13 +274,29 @@ jobs:
...
+        $file = Get-Item "CoseSignTool.dll"
+        $version = $file.VersionInfo.FileVersion
+        Write-Output "File Version is $version"
+      shell: pwsh
+      working-directory: ./published/debug
 
     # Copy the docs, license, and markdown files to the release folders.
     - name: Copy docs to release folders
       run: |
+        exit 1  # remove this line when the script is ready
```
Comment (mayuram2) [inline .github/workflows/dotnet.yml:299]: Should this be removed?

Comment (lemccomb) [reply .github/workflows/dotnet.yml:299]: Yes!

Comment (lemccomb) [reply .github/workflows/dotnet.yml]: done

**Review by JoeBussell — APPROVED** [review]

**Pull request merged**